test/sources/pipenv_test.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+require "test_helper"
+require "tmpdir"
+
+if Licensed::Shell.tool_available?("pipenv")
+  describe Licensed::Sources::Pipenv do
+    let (:fixtures) { File.expand_path("../../fixtures/pipenv", __FILE__) }
+    let (:config)   { Licensed::Configuration.new }
+    let (:source)   { Licensed::Sources::Pipenv.new(config) }
+
+    describe "enabled?" do
+      it "is true if pipenv source is available" do
+        Dir.chdir(fixtures) do
+          assert source.enabled?
+        end
+      end
+
+      it "is false if pipenv source is not available" do
+        Dir.mktmpdir do |dir|
+          Dir.chdir(dir) do
+            refute source.enabled?
+          end
+        end
+      end
+    end
+
+    describe "dependencies" do
+      it "detects top-level dependencies" do
+        Dir.chdir fixtures do
+          dep = source.dependencies.detect { |d| d.name == "pylint" }
+          assert dep
+          assert_equal "pipenv", dep.record["type"]
+          assert dep.record["homepage"]
```

Comment:
can you also add some verifications to these tests that `dep.version` is the expected value?

Reply:
Good point, added in next commit.